bundle/src/main/groovy/org/cid15/aem/groovy/console/servlets/ReplicateScriptServlet.groovy

```diff
@@ -0,0 +1,71 @@
+package org.cid15.aem.groovy.console.servlets
+
+import com.day.cq.commons.jcr.JcrConstants
+import com.google.common.base.Charsets
+import com.google.common.net.MediaType
+import groovy.util.logging.Slf4j
+import org.apache.commons.io.IOUtils
+import org.apache.sling.api.SlingHttpServletRequest
+import org.apache.sling.api.SlingHttpServletResponse
+import org.apache.sling.api.resource.ResourceResolverFactory
+import org.cid15.aem.groovy.console.configuration.ConfigurationService
+import org.jetbrains.annotations.NotNull
+import org.osgi.service.component.annotations.Component
+import org.osgi.service.component.annotations.Reference
+
+import javax.servlet.Servlet
+import javax.servlet.ServletException
+
+import static javax.servlet.http.HttpServletResponse.SC_BAD_REQUEST
+import static javax.servlet.http.HttpServletResponse.SC_FORBIDDEN
+import static org.cid15.aem.groovy.console.constants.GroovyConsoleConstants.CHARSET
+import static org.cid15.aem.groovy.console.constants.GroovyConsoleConstants.PATH_REPLICATION_FOLDER
+import static org.cid15.aem.groovy.console.constants.GroovyConsoleConstants.SCRIPT
+
+@Component(service = Servlet, immediate = true, property = [
+        "sling.servlet.paths=/bin/groovyconsole/replicate"
+])
+@Slf4j("LOG")
+class ReplicateScriptServlet extends AbstractJsonResponseServlet {
+    @Reference
+    private ConfigurationService configurationService
+
+    @Reference
+    private ResourceResolverFactory resourceResolverFactory;
+
+    @Override
+    protected void doPost(@NotNull SlingHttpServletRequest request, @NotNull SlingHttpServletResponse response) throws ServletException, IOException {
+        if (configurationService.hasPermission(request)) {
+            def script = request.getRequestParameter(SCRIPT)?.getString(Charsets.UTF_8.name())
+            if (script) {
+                LOG.info("Replicate script")
```

Comment:
debug, maybe you can add trace logging about the actual script

Reply:
true